src/main/java/seedu/address/logic/commands/view/ViewApplicationCommand.java

```diff
@@ -48,11 +53,21 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(String.format(MESSAGE_INVALID_ITEM_DISPLAYED_INDEX, APPLICATION_NAME));
         }
 
-        // TODO after model/ui methods are done: Check if current item is already in view
-        // TODO: Set model to show view
+        String resultMessage = messageViewSuccess;
+        boolean shouldSwitchTab = false;
+        boolean shouldSwitchDisplay = true;
+        if (model.getTabName() != TabName.APPLICATION) {
+            model.setTabName(TabName.APPLICATION);
+            shouldSwitchTab = true;
+        } else if (model.getViewIndex().equals(targetIndex)) {
+            resultMessage = messageAlreadyViewing;
+            shouldSwitchDisplay = false;
+        }
 
-        String viewSuccessMessage = String.format(MESSAGE_VIEW_SUCCESS, APPLICATION_NAME, targetIndex);
-        return getCommandResult(model, viewSuccessMessage, TabName.APPLICATION);
+        if (shouldSwitchDisplay) {
```

Comment:
just a minor question but is there a need for this if block? I think we could just shift it into the if-else block. However, if this is a better coding standard, then I think we can stick to this.

Reply:
I can't shift it straight into the current if-else block due to the nature of the condition checking. If we want to make it all part of one block, here's an option:
```
if (model.getTabName() != TabName.APPLICATION) {
    model.setTabName(TabName.APPLICATION);
    shouldSwitchTab = true;
    model.setViewIndex(targetIndex);
} else if (model.getViewIndex().equals(targetIndex)) {
    resultMessage = messageAlreadyViewing;
    shouldSwitchDisplay = false;
} else {
    model.setViewIndex(targetIndex);
}
```
But this means `model.setViewIndex(targetIndex)` appears twice.

Another option is to simply remove the if condition and do:
```
if (model.getTabName() != TabName.APPLICATION) {
    model.setTabName(TabName.APPLICATION);
    shouldSwitchTab = true;
    model.setViewIndex(targetIndex);
} else if (model.getViewIndex().equals(targetIndex)) {
    resultMessage = messageAlreadyViewing;
    shouldSwitchDisplay = false;
}

model.setViewIndex(targetIndex);
```
But this means that even when the model's view index is already the index that the user inputs, we still set it anyway (unnecessary call to setViewIndex). It won't affect the program though.

Between the current code and these 2 options, which do yall think is better? Or are there other alternatives yall may have?

Reply:
oh if this the case, I believe the current code is a better option

Reply:
I think the current code is better. And also see my comment below too

Reply:
Ah good question! I raised this in our Tele group last night, but basically how the view index works is that it takes the index from the tab that is currently in view. So for example, if a user is on say the Application tab and he is viewing the 3rd Application, if he executes `view com 3`, if we do not check for the tab first, then the program will simply think we're already viewing the 3rd item so it doesn't need to do anything, which is wrong of course. That said, now that you've raised this point, I believe I may be able to abstract away the first if statement, but I'll definitely need that condition check in the else-if statement that checks for if the user is on the same tab and currently viewing the indexed item. So I can minimize the tab checking but we need that one minimally.

Reply:
Oh okay thanks for the clarification!

Reply:
No problem, thanks for highlighting! Have made the changes!